File: hw/robalino/u2/Hw 18 Inheritance/cage.py
from typing import List


class Cage:
    def __init__(self, id: int, animals: List['Animal']):
        self._id = id
        self._animals = animals

    def __str__(self) -> str:
        return f"Cage(id={self._id}, animals={self._animals})"

    @property
    def id(self) -> int:
        return self._id

    @id.setter
    def id(self, value: int):
        self._id = value

    @property
    def animals(self) -> List['Animal']:
        return self._animals

    @animals.setter
    def animals(self, value: List['Animal']):
        self._animals = value

File: hw/pilatasig/u2/ZooApp JavaScript/ec/edu/espe/zoo/model/Cage.js
export class Cage {
    constructor(id, animals = []) {
        this._id = id; // int
        this._animals = animals; // Array of Animal objects
    }

    // Getter para id
    get id() {
        return this._id;
    }

    // Setter para id
    set id(value) {
        this._id = value;
    }

    // Getter para animals
    get animals() {
        return this._animals;
    }

    // Setter para animals
    set animals(value) {
        this._animals = value;
    }

    // Método toString
    toString() {
        const animalsString = this._animals.map(animal => animal.toString()).join(", ");
        return `Cage{id=${this._id}, animals=[${animalsString}]}`;
    }
}
